src/biokbase/narrative/tests/test_viewers.py

```diff
@@ -1,148 +1,142 @@
-        with pytest.raises(ValueError, match="not a compatible data type"):
-            viewers.view_as_clustergrammer(self.attribute_set_ref)
-
-    def test__get_categories(self):
-        import pandas as pd
-        from biokbase.narrative import viewers
-
-        ids = ["WRI_RS00010_CDS_1", "WRI_RS00015_CDS_1", "WRI_RS00025_CDS_1"]
-        mapping = {
-            "WRI_RS00010_CDS_1": "test_row_instance_1",
-            "WRI_RS00015_CDS_1": "test_row_instance_2",
-            "WRI_RS00025_CDS_1": "test_row_instance_3",
-        }
-        index = [
-            (
-                "WRI_RS00010_CDS_1",
-                "test_attribute_1: 1",
-                "test_attribute_2: 4",
-                "test_attribute_3: 7",
-            ),
-            (
-                "WRI_RS00015_CDS_1",
-                "test_attribute_1: 2",
-                "test_attribute_2: 5",
-                "test_attribute_3: 8",
-            ),
-            (
-                "WRI_RS00025_CDS_1",
-                "test_attribute_1: 3",
-                "test_attribute_2: 6",
-                "test_attribute_3: 9",
-            ),
-        ]
-        filtered_index = [
-            ("WRI_RS00010_CDS_1", "test_attribute_1: 1"),
-            ("WRI_RS00015_CDS_1", "test_attribute_1: 2"),
-            ("WRI_RS00025_CDS_1", "test_attribute_1: 3"),
-        ]
-        multi_index = pd.MultiIndex(
-            levels=[
-                ["WRI_RS00010_CDS_1", "WRI_RS00015_CDS_1", "WRI_RS00025_CDS_1"],
-                ["1", "2", "3"],
-            ],
-            labels=[[0, 1, 2], [0, 1, 2]],
-            names=["ID", "test_attribute_1"],
-        )
-        assert ids, viewers._get_categories(ids == self.generic_ref)
-        with pytest.raises(ValueError, match="not in the provided mapping"):
-            viewers._get_categories(["boo"], self.generic_ref, self.attribute_set_ref, mapping)
-        with pytest.raises(ValueError, match="has no attribute"):
-            viewers._get_categories(["boo"], self.generic_ref, self.attribute_set_ref)
-        assert index == viewers._get_categories(
-            ids,
-            self.generic_ref,
-            self.attribute_set_ref,
-            mapping,
-            clustergrammer=True,
-        )
-
-        pd.testing.assert_index_equal(
-            multi_index,
-            viewers._get_categories(
-                ids,
-                self.generic_ref,
-                self.attribute_set_ref,
-                mapping,
-                {"test_attribute_1"},
-            ),
-        )
-        assert filtered_index == viewers._get_categories(
+from biokbase.narrative import viewers
+
+ATTRIBUTE_SET_REF = "36095/73/1"
+GENERIC_REF = "36095/74/1"
+EXPRESSION_MATRIX_REF = "28852/11/1"
+
+
+@pytest.mark.parametrize("arg_type", ["col_categories", "row_categories", "normalize_on"])
+def test_bad_view_as_clustergrammer_params_generic_ref(arg_type: str) -> None:
```

Comment:
The existing tests were all skipped and they used that lovely testing paradigm "one test method, lots of tests". I split them out into separate named tests.